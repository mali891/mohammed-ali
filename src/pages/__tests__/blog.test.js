import React from 'react'
import { render } from '@testing-library/react'

import Blog from '../blog'

const props = {
  data: {
    allMarkdownRemark: {
      edges: [
        {
          node: {
            frontmatter: {
              title: 'Test title 1',
              path: '/test-title/1',
              date: '2020-01-01',
              length: '3 min'
            }
          }
        },
        {
          node: {
            frontmatter: {
              title: 'Test title 2',
              path: '/test-title/2',
              date: '2020-02-02',
              length: '4 min'
            }
          }
        }
      ]
    }
  }
}

describe('Blog', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<Blog {...props} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
