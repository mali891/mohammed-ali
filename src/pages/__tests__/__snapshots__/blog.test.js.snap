// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog renders correctly 1`] = `
<DocumentFragment>
  <div
    class="u-anim--fade-in"
  >
    <header
      class="c-header"
      css="[object Object]"
    >
      <div
        class="c-header__container c-container"
        css="[object Object]"
      >
        <h1
          class="c-heading c-heading--xl c-heading--no-margin c-heading--align-center"
          css="[object Object]"
        >
          Blog
        </h1>
      </div>
      <button
        class="c-header__more-link c-link u-clr--secondary"
        css="[object Object]"
      >
        More
      </button>
    </header>
    <div
      class="c-header-scroll"
      css="[object Object]"
    />
    <main
      class="c-main"
    >
      <section
        class="c-content-group"
        css="[object Object]"
      >
        <div
          class="c-card-container c-card-container--margin-bottom"
          css="[object Object]"
        >
          <a
            class="c-card u-bg--tertiary c-card--link c-card-fullheight c-link u-clr--secondary c-link--no-hover"
            css="[object Object]"
            href="/test-title/1"
          >
            <div
              class="c-card--fullheight"
            >
              <div
                class="c-card__heading"
              >
                <span
                  class="c-heading c-heading--sm c-heading--no-margin"
                  css="[object Object]"
                >
                  Test title 1
                </span>
              </div>
              <div
                class="c-card__content"
              >
                <span>
                  1st Jan 2020
                </span>
                <span>
                  3 min read
                </span>
              </div>
            </div>
          </a>
          <a
            class="c-card u-bg--tertiary c-card--link c-card-fullheight c-link u-clr--secondary c-link--no-hover"
            css="[object Object]"
            href="/test-title/2"
          >
            <div
              class="c-card--fullheight"
            >
              <div
                class="c-card__heading"
              >
                <span
                  class="c-heading c-heading--sm c-heading--no-margin"
                  css="[object Object]"
                >
                  Test title 2
                </span>
              </div>
              <div
                class="c-card__content"
              >
                <span>
                  2nd Feb 2020
                </span>
                <span>
                  4 min read
                </span>
              </div>
            </div>
          </a>
        </div>
        <div
          class="c-card-container c-card-container--margin-bottom"
          css="[object Object]"
        />
      </section>
    </main>
  </div>
</DocumentFragment>
`;
